apps/api/src/applicant/ienapplicant.service.ts

```diff
@@ -231,14 +240,16 @@ export class IENApplicantService {
 
     data.status = status_obj;
 
-    let job = null;
+    // needed specfic typing as job is now being used inside nested transaction
+    let job = new IENApplicantJob();
+
```

Comment:
how about this? ienapplicantUtilService.getJob should handle the case that job_id is not defined or throw an exception.
```
const job = await this.ienapplicantUtilService.getJob(job_id);
if (id !== job?.applicant.id) {
  throw new BadRequestException('Provided applicant and competition/job does not match');
}
```

Reply:
fixed